frontend/public/components/catalog-items.jsx

```diff
@@ -1,48 +1,204 @@
-    vendor={item.tileProvider ? `Provided by ${item.tileProvider}` : null}
-    description={item.tileDescription}
-  />;
-  return item.href ? <Link className="co-catalog-item-tile" to={item.href}>{catalogTile}</Link> : catalogTile;
-};
+export class CatalogTileViewPage extends React.Component {
+  constructor(props) {
+    super(props);
 
-CatalogItem.displayName = 'CatalogItem';
-
-CatalogItem.propTypes = {
-  item: PropTypes.shape({
-    obj: PropTypes.object.isRequired,
-    kind: PropTypes.string.isRequired,
-    tileName: PropTypes.string.isRequired,
-    tileIconClass: PropTypes.string,
-    tileImgUrl: PropTypes.string,
-    tileDescription: PropTypes.string.isRequired,
-    tileProvider: PropTypes.string,
-  }).isRequired
-};
+    this.state = {
+      showAllItemsForCategory: null,
+      activeTabs: ['all'], // array of tabs [main category, sub-category]
+    };
+
+    if (props.items) {
+      this.allCategories = categorizeItems(props.items);
+      _.assign(this.state, {
+        currentCategories: this.allCategories,
+        numItems: _.first(this.allCategories).numItems,
+      });
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    const {items} = this.props;
+
+    if (items !== nextProps.items) {
+      this.allCategories = categorizeItems(items);
+      this.setState({
+        currentCategories: this.allCategories,
+        numItems: _.first(this.allCategories).numItems
+      });
+    }
+  }
+
+  isAllTabActive() {
+    const { activeTabs } = this.state;
+    return activeTabs[0] === 'all';
+  }
+
+  activeTabIsSubCategory(subCategories) {
+    const { activeTabs } = this.state;
+    return activeTabs.length !== 2 ? false : _.find(subCategories, ['id', activeTabs[1]]) !== undefined;
+  }
+
+  isActiveTab(categoryId) {
+    const { activeTabs } = this.state;
+    return activeTabs.length === 2 ? categoryId === activeTabs[1] : categoryId === activeTabs[0];
+  }
+
+  doesHaveActiveDescendants(categoryId) {
+    const { activeTabs } = this.state;
+    return activeTabs[0] === categoryId;
+  }
+
+  renderTabs(category, parentCategoryId=null){
+    if (!category.numItems) {
+      return;
+    }
+    return <VerticalTabs.Tab
+      key={category.id}
+      title={category.label}
+      active={this.isActiveTab(category.id)}
+      onActivate={() => this.onActivateTab(parentCategoryId ? [parentCategoryId, category.id] : [category.id])}
+      hasActiveDescendant={this.doesHaveActiveDescendants(category.id)}
+      shown={category.id === 'all'}>
+      {category.subCategories && category.numItems && <VerticalTabs key={`${category.id}-tabs`}
+        restrictTabs={true} activeTab={this.activeTabIsSubCategory(category.subCategories)}>
+        {_.map(category.subCategories, subCategory => this.renderTabs(subCategory, category.id))}
+      </VerticalTabs>
+      }
+    </VerticalTabs.Tab>;
+  }
+
+  renderCategoryTabs() {
+    return (
+      <VerticalTabs restrictTabs={true} activeTab={this.isAllTabActive()} shown="true">
```

Comment:
don't need shown="true" here

Reply:
activeTab should be true if any of the top level categories or All, is active

Reply:
can replace `restrictTabs={true} ` with simply `restrictTabs`

Reply:
> activeTab should be true if any of the top level categories or All, is active

I believe when I had that, all top level tabs displayed even when drilled down into sub-categories.  I believe we only want to show the active 'tree' of the active tab, not other parent/main categories.

Reply:
Do we really want to hide the other top level categories? That seems strange to me. There aren't that many.

Reply:
That is the design, @tlwu2013

Reply:
That is the design, @tlwu2013?

Reply:
@tlwu2013, @serenamarie125, we do need to decide on this.  IMO it does seem a little jarring to hide top level categories when drilling down to sub-cats, plus it requires and extra click on 'All Cats' to bring back the top level categories.

Reply:
It also moves the position of the category you just clicked, which is a little annoying if you wanted to click a subcategory. They don't expand below your mouse cursor where you'd expect.